index.js

```diff
@@ -36,6 +36,7 @@ module.exports = function gulpPug(options) {
         }
         file.contents = new Buffer(compiled);
       } catch (e) {
+        e.filename = opts.filename;
```

Comment:
This should use the [`fileName`](https://github.com/gulpjs/plugin-error/blob/master/index.d.ts#L43) option for `PluginError` instead of mutating the cause.

`new PluginError('gulp-pug', e, {fileName: opts.filename})`